combat/CombatSystem.cpp

```diff
@@ -889,10 +889,18 @@ namespace {
             // do actual attacks
             Attack(attacker, *weapon_it, target, combat_state.combat_info, bout, round);
 
-            // mark attackers as valid targets for attacked object's owners, so
-            // attacker they can be counter-attacked in subsequent rounds if it
+            // mark attacker as valid target for attacked object's owner, so that regardless
+            // of visibility the attacker can be counter-attacked in subsequent rounds if it
             // was not already attackable
             combat_state.empire_infos[target->Owner()].target_ids.insert(attacker->ID());
+            // Also ensure that attacker (and their fleet if attacker was a ship) are
+            // revealed with at least BASIC_VISIBILITY to the taget empire
+            combat_state.combat_info.empire_object_visibility[target->Owner()][attacker->ID()] = 
+                    std::max(GetUniverse().GetObjectVisibilityByEmpire(attacker->ID(), target->Owner()), VIS_BASIC_VISIBILITY);
```

Comment:
Why is it necessary to poll the main universe visibility info here? Doesn't the container you're modifying have the relevant info about the best visibility level empires have for each object?